Hazelcast.Net/Hazelcast.Client.Connection/ClientConnection.cs

```diff
@@ -77,15 +77,16 @@ internal sealed class ClientConnection
             {
                 _clientSocket = new Socket(isa.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
                 _clientSocket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.KeepAlive, true);
-
-                var lingerOption = new LingerOption(true, 5);
+                var lingerOption = new LingerOption(true, socketOptions.GetLingerSeconds());
                 if (socketOptions.GetLingerSeconds() > 0)
                 {
-                    lingerOption.LingerTime = socketOptions.GetLingerSeconds();
+                    _clientSocket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.Linger, lingerOption);
+                }
+                else
+                {
+                    _clientSocket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.DontLinger, lingerOption);
```

Comment:
you may consider changing with _clientSocket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.DontLinger, new LingerOption(true, 0)); Currently I understand that the lingerOption will be constructed with negative linger seconds, which most probably have the same affect as described here https://msdn.microsoft.com/en-us/library/system.net.sockets.lingeroption.lingeroption(v=vs.110).aspx but it will be more clear this way.